lib/elixir/lib/task.ex

```diff
@@ -701,11 +743,12 @@ defmodule Task do
     end
   end
 
-  defp yield_many([%Task{ref: ref, owner: owner} = task | rest], timeout_ref, timeout) do
-    if owner != self() do
-      raise ArgumentError, invalid_owner_error(task)
-    end
+  defp yield_many([%Task{owner: owner} = task | _rest], _timeout_ref, _timeout)
+       when owner != self() do
+    raise ArgumentError, invalid_owner_error(task)
+  end
```

Comment:
I think this is better with just one clause so we do the whole pattern matching on the head just once. If we really want to go this way, I would invert the clauses and go with `when owner == self()` first and then fall back to the error case otherwise.

Reply:
Yes, let's keep the conditional here. :+1:

Reply:
Great, I have taken it a step further and apply this in other places